SleepEngine/Engine/Components/HierarchyNode.cpp

```diff
@@ -58,17 +80,15 @@ void HierarchyNode::detachFromParent()
     }
 
     auto* const parentNode = m_parent->getComponent<HierarchyNode>();
-    parentNode->removeChild(m_parent);
+    parentNode->removeChild(m_object);
     m_parent = nullptr;
 }
 
 void HierarchyNode::update(float dt)
 {
-    Base::update(dt);
-
-    for (auto& childNodeObject : m_children)
+    for (auto* const child : m_children)
     {
-        childNodeObject->update(dt);
+        child->update(dt);
     }

```

Comment:
So HierarchyNode still updates children but doesn't own it now. EntityManager knows that it shouldn't update these objects.